gwpy/cli/cliproduct.py

```diff
@@ -724,6 +730,50 @@ def run(self):
                 show_error = False
             self.plot_num += 1
 
+    def add_segs(self, args):
+        """ If requested add DQ segments"""
+        std_segments = \
+            [
+                '{ifo}:DMT-GRD_ISC_LOCK_NOMINAL:1',
+                '{ifo}:DMT-DC_READOUT_LOCKED:1',
+                '{ifo}:DMT-CALIBRATED:1',
+                '{ifo}:DMT-ANALYSIS_READY:1'
+            ]
+        segments = list()
+        if hasattr(args, 'std_seg'):
+            if args.std_seg:
+                segments = std_segments
+            if args.seg:
+                for seg in args.seg:
+                    # NB: args.seg may be list of lists
+                    segments += seg
+
+            chan = args.chan[0][0]
+            m = re.match('^([A-Za-z][0-9]):', chan)
+            ifo = '?:'
+            if m:
+                ifo = m.group(1)
+            start = None
+            for ts in self.timeseries:
+                if start:
+                    start = min(ts.t0, start)
+                    end = max(ts.t0+ts.duration, end)
+                else:
+                    start = ts.t0
+                    end = start + ts.duration
+
+            for segment in segments:
+                seg_name = segment.replace('{ifo}', ifo)
+                try:
+                    seg_data = DataQualityFlag. \
+                        query_dqsegdb(seg_name, start, end,
+                                      url='https://segments.ligo.org')
+                except:             # noqa: E722
```

Comment:
Can you please itemise the exceptions that we don't care about here, rather than leaving a bare `except`?

Reply:
The point is we don't care why. If segments fail we try segments-backup.
Is there a way to get a complete list of possible exceptions?

Reply:
If segmemts-backup also fails we report that error.
I'd prefer to leave it unless we can get a list of exceptions.

Reply:
I presume we only care about failed connections, and not missing flags etc. in that case we should probably use `except URLError` or similar.

Reply:
We can do that but isn't it possible that backup succeeds when segments fail for other than connection problems?  If the error is unrecoverable won't we get the same error from backup?  In my mind it only costs one more connection on the rare error.

Reply:
This is resolved by your suggestion to remove segdb urls from this level